modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpCache.java

```diff
@@ -56,18 +62,27 @@ <T extends AbstractResponse> T putIfAbsent(
     ) {
         // can't use cache.computeIfAbsent due to the elevated permissions for the jackson (run via the cache loader)
         CacheKey cacheKey = new CacheKey(ip, databasePath);
+        long cacheStart = System.nanoTime();
         // intentionally non-locking for simplicity...it's OK if we re-put the same key/value in the cache during a race condition.
         AbstractResponse response = cache.get(cacheKey);
+        long cacheRequestTime = System.nanoTime() - cacheStart;
 
         // populate the cache for this key, if necessary
         if (response == null) {
+            long start = System.nanoTime();
             response = retrieveFunction.apply(ip);
+            long databaseRequestTime = System.nanoTime() - start;
+            storeQueryTimeInNanos.addAndGet(databaseRequestTime);
+            missesTimeInNanos.addAndGet(cacheRequestTime);
+            missesTimeInNanos.addAndGet(databaseRequestTime);
             // if the response from the database was null, then use the no-result sentinel value
             if (response == null) {
                 response = NO_RESULT;
             }
             // store the result or no-result in the cache
             cache.put(cacheKey, response);
```

Comment:
The miss time should include the `cache.put`, IMHO.

Reply:
Oh good point. Think I ought to have a separate counter for that alone, too? Right now you can compute the cache query time with (missesTimeInNanos - storeQueryTimeInNanos). But if I include the cache put time, you can no longer do that. You need another counter in order to do that (either cache_put_time_in_millis or cache_query_time_in_millis).

Reply:
Indeed, you lose the specificity of being able to determine the 'cache query time' but you'd still have a more general 'cache overhead time', right? I'm not sure how much I think the distinction is going to be useful to **users** -- if we're trying to optimize this code we be looking at a profiler for the real nitty-gritty, rather than these stats...

Reply:
👍